services/fileuploader/gzwriter.go

```diff
@@ -0,0 +1,52 @@
+package fileuploader
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type MultiFileWriter interface {
+	Write(workspaceID, path string, data []byte) (int, error)
+	Close() (map[string]string, error)
+}
+
+type GZFileHandler struct {
+	gzWriters map[string]misc.GZipWriter
+	dumps     map[string]string
+}
+
+func NewGzWriter() *GZFileHandler {
+	return &GZFileHandler{
+		gzWriters: make(map[string]misc.GZipWriter),
+		dumps:     make(map[string]string),
+	}
+}
+
+func (g *GZFileHandler) Write(workspaceID, path string, data []byte) (count int, err error) {
+	if _, ok := g.gzWriters[workspaceID]; !ok {
+
+		err := os.MkdirAll(filepath.Dir(path), os.ModePerm)
+		if err != nil {
+			return 0, fmt.Errorf("creating gz file %q: %w", path, err)
+		}
+
+		g.gzWriters[workspaceID], err = misc.CreateGZ(path)
+		g.dumps[workspaceID] = path
+		if err != nil {
+			return 0, err
```

Comment:
We could use some context error here considering the above.

Reply:
Sorry, you want me to check if context has been cancelled here?